src/ProjectNeon/Assets/Features/Effects/ForNumberOfTurns.cs

```diff
@@ -1,38 +1,32 @@
 ﻿
 /**
  * Wraps the execution of an effect so it is only executed for a certain amount of turns.
+ * 
+ * Turn ending decrease the remainingDuration in one. 
  */
-public class ForNumberOfTurns : ITemporalState, Effect
+public class ForNumberOfTurns : Effect
 {
     
     private int _remainingDuration;
 
     private Effect _effect;
 
-    private bool _isDebuff;
+    public bool IsActive => _remainingDuration >= 0;
 
-    public IStats Stats { get; } = new StatAddends();
-    public bool IsDebuff => _isDebuff;
-    public bool IsActive => _remainingDuration > 0;
-
-    public void AdvanceTurn()
+    private void AdvanceTurn()
     {
-        if (_remainingDuration <= 0) return;
-
-        _remainingDuration--;
-        
+        if (_remainingDuration < 0) return;
+            _remainingDuration--;
     }
 
-    public ForNumberOfTurns(Effect effect, int duration, bool isDebuff)
+    public ForNumberOfTurns(Effect effect, int duration)
     {
         _effect = effect;
         _remainingDuration = duration;
-        _isDebuff = isDebuff;
+        BattleEvent.Subscribe<TurnEnd>((turnEnd) => AdvanceTurn(), this);
```

Comment:
All Event Subscriptions must have Unsubscriptions.

Reply:
@SilasReinagel `Unsubscriptions` weren't needed until the last PR. We don't have them for any other classes that uses messaging subscription (`Attack`, `EffectOnAttacked`, EffectOnAttacker` etc,). Where should I put it?

Reply:
Well, for things that naturally expires, they should unsubscribe themselves once finished. Then the garbage collector can clean them up. 

Mentally, you should see a red flag any time there is no unsubscribe for any subscription.

I just ran into some program crashing bugs in another project due to missing unsubscriptions, so now I am being more vigilant about them.

Reply:
@SilasReinagel Okay, fil add them to the related events